src/main/resources/assets/computercraft/shaders/monitor.frag

```diff
@@ -24,17 +24,15 @@ void main() {
     ivec2 cell = ivec2(corner);
     int index = 3 * (clamp(cell.x, 0, u_width - 1) + clamp(cell.y, 0, u_height - 1) * u_width);
 
-    if(cell.x < 0 || cell.y < 0 || cell.x >= u_width || cell.y >= u_height) {
-        int bg = int(texelFetch(u_tbo, index + 2).r * 255.0);
-        gl_FragColor = vec4(u_palette[bg], 1.0);
-        return;
-    }
+    // 1 if 0 <= x, y < width, height, 0 otherwise
+    vec2 outside = step(ivec2(0, 0), cell) * step(cell, ivec2(u_width - 1, u_height - 1));
```

Comment:
Why does `step` return a `vec2` instead of an `ivec2`? Is this an intel bug, or actual behaviour?

Reply:
In my interpretation of the docs, it should return `ivec2`, but don't take my word for it. GPUs are glorified floating point processing machines, so I wouldn't be surprised if it doesn't.

Reply:
Fun facts! It only operates on floats. Or at least it does on NVIDIA.